src/Compiler/Checking/CheckPatterns.fs

```diff
@@ -35,6 +35,24 @@ type cenv = TcFileState
 // Helpers that should be elsewhere
 //-------------------------------------------------------------------------
 
+// CAUTION: This function operates over the untyped tree, so should be used only when absolutely necessary. It doesn't verify assembly origine nor does it respect type aliases.
+let inline findSynAttribute (attrName: string) (synAttrs: SynAttributes) =
```

Comment:
I think this is now a lot simpler than isAutoOpenAttribute.

If this is moved e.g. to SyntaxTreeOps, it could then be referenced from CheckPatterns as well as from TrieMapping.fs, which is doing the same for AutoOpen attribute, lines 18-46.

If this unificaiton is done, I prefer the double List.exists compared to maintaining a mutable boolean.

`   attributes|> List.exists (fun (atl: SynAttributeList) -> List.exists ...isDesiredAttribute... atl.Attributes)`